src/packer.js

```diff
@@ -0,0 +1,105 @@
+const Lang = imports.lang;
+
+const Gio = imports.gi.Gio;
+const GLib = imports.gi.GLib;
+
+const Index = imports.index;
+const Links = imports.links;
+const Shard = imports.shard;
+
+const Packer = new Lang.Class ({
+    Name: 'Packer',
+
+    _init: function (json_path, shard_path) {
+        let json_file = Gio.File.new_for_path(json_path);
+        let [success, data, tag] = json_file.load_contents(null);
+        this._json = JSON.parse(data);
+
+        this._shard = new Shard.Shard(shard_path);
+        this._index = new Index.Index();
+        this._links = new Links.Links();
+    },
+
+    run: function () {
+        print('Writing articles');
+        this._json['articles'].forEach((entry) => {
```

Comment:
optional: `this._json['articles'].forEach(this._add_article, this);` is more concise but arguably less readable